tf/net.py

```diff
@@ -412,18 +499,24 @@ def get_weights_v2(self, names):
                 # headcount is set with set_headcount()
                 continue
 
-            pb_name, block, pol_encoder_block = self.tf_name_to_pb_name(name)
+            pb_name, block, pol_encoder_block, encoder_block = self.tf_name_to_pb_name(name)
 
             if pb_name is None:
                 raise ValueError(
                     "Don't know where to store weight in protobuf: {}".format(
                         name))
 
             if block is None:
-                if pol_encoder_block is None:
-                    pb_weights = self.pb.weights
+                if name.startswith("policy"):
```

Comment:
I think a given weight can only have at most one of pol_encoder_block or encoder_block set to not None, so you can probably just nest the checks rather than needing to check the name.